app/models/miq_worker/runner.rb

```diff
@@ -450,23 +450,22 @@ def clean_broker_connection
   def ruby_object_usage
     types = Hash.new { |h, k| h[k] = Hash.new(0) }
     ObjectSpace.each_object do |obj|
-      types[obj.class][:count] += 1
-      next if obj.kind_of?(DRbObject) || obj.kind_of?(WeakRef)
-      if obj.respond_to?(:length)
-        len = obj.length
-        if len.kind_of?(Numeric)
-          types[obj.class][:max]    = len if len > types[obj.class][:max]
-          types[obj.class][:total] += len
-        end
-      end
+      types[obj.class.name][:count] += 1
     end
     types
   end
 
+  LOG_RUBY_OBJECT_USAGE_INTERVAL = 60
   def log_ruby_object_usage(top = 20)
-    if top > 0
+    return unless top > 0
+
+    t = Time.now.utc
+    @last_ruby_object_usage ||= t
+
+    if (@last_ruby_object_usage + LOG_RUBY_OBJECT_USAGE_INTERVAL) < t
       types = ruby_object_usage
-      $log.info("Ruby Object Usage: #{types.sort_by { |_klass, h| h[:count] }.reverse[0, top].inspect}")
+      $log.info("Ruby Object Usage: #{types.sort_by(&:count)[0, top].inspect}")
```

Comment:
It also takes the bottom n instead of the top n because it doesn't do the reverse.